### DISCUSSION TITLE
External ISPN host configuration (at runtime)

### DISCUSSION THREAD

**Discussion by avanlangup** [keycloak/keycloak#22308] [Q&A] [answered]
We define some remote distributed cache in cache-ispn.xml and build KC 21.1.1 docker image with  KC_CACHE_CONFIG_FILE=cache-ispn.xml 

The problem is that the authentication should not be hard coded in clear in git repository

```
<remote-server host="apps-infinispan.ns-dev.svc.cluster.local" port="11222" />
      <security>
          <authentication server-name="infinispan">
              <digest username="infinispan" password="xxx" realm="default"/>
          </authentication>
      </security>
```

How can we inject env variables like host and password (secret helm values) at deployment time ?

Thanks for your advice

note:  If we do not use _--optimized_ to start KC, we will lose all the previous configuration files

**Comment by mhajas** [comment, marked as answer]
You can have a look at `keycloak-benchmark` repository. Here is how we configure it there:

ISPN config file:
https://github.com/keycloak/keycloak-benchmark/blob/dd609924afdb88c8ae6f2d5b89cf9a4d96f04d63/provision/minikube/keycloak/config/kcb-infinispan-cache-remote-store-config.xml#L70-L81

Keycloak operator config:
https://github.com/keycloak/keycloak-benchmark/blob/dd609924afdb88c8ae6f2d5b89cf9a4d96f04d63/provision/minikube/keycloak/templates/keycloak.yaml#L109-L122
